File: src/main/java/me/zeroeightsix/kami/module/modules/misc/XCarry.java
package me.zeroeightsix.kami.module.modules.misc;

import me.zero.alpine.listener.EventHandler;
import me.zero.alpine.listener.Listener;
import me.zeroeightsix.kami.event.events.PacketEvent;
import me.zeroeightsix.kami.module.Module;
import net.minecraft.network.play.client.CPacketCloseWindow;

/**
 * @author Hamburger2k
 */
@Module.Info(name = "XCarry", category = Module.Category.PLAYER, description = "Store items in crafting slots", showOnArray = Module.ShowOnArray.OFF)
public class XCarry extends Module {
    @EventHandler
    private Listener<PacketEvent.Send> l = new Listener<>(event -> {
        if (event.getPacket() instanceof CPacketCloseWindow) {
            event.cancel();
        }
    });
}

File: src/main/java/me/zeroeightsix/kami/module/modules/player/XCarry.kt
package me.zeroeightsix.kami.module.modules.player

import me.zeroeightsix.kami.event.events.PacketEvent
import me.zeroeightsix.kami.module.Module
import org.kamiblue.event.listener.listener
import net.minecraft.network.play.client.CPacketCloseWindow

@Module.Info(
        name = "XCarry",
        category = Module.Category.PLAYER,
        description = "Store items in crafting slots"
)
object XCarry : Module() {
    init {
        listener<PacketEvent.Send> {
            if (it.packet is CPacketCloseWindow) it.cancel()
        }
    }
}